src/T2DMap.cpp

```diff
@@ -4319,25 +4405,50 @@ void T2DMap::slot_setCustomLine()
         pI->setTextAlignment(2, Qt::AlignLeft);
     }
 
-    b_ = d->findChild<QPushButton*>("button_cancel");
-    if (!b_) {
-        qWarning(R"(T2DMap::slot_setCustomLine() ERROR: failed to find "cancel" button!)");
+    QGroupBox *pGroupBox_setting = d->findChild<QGroupBox*>(QStringLiteral("groupBox_settings"));
+    QGroupBox *pGroupBox_normalExits = d->findChild<QGroupBox*>(QStringLiteral("groupBox_normalExits"));
+    QGroupBox *pGroupBox_specialExits = d->findChild<QGroupBox*>(QStringLiteral("groupBox_specialExits"));
+
+    if (!pGroupBox_setting || !pGroupBox_normalExits || !pGroupBox_specialExits) {
+        qWarning("T2DMap::slot_setCustomLine() ERROR: failed to find a groupBox item!");
         return;
     }
-    connect(b_, SIGNAL(clicked()), d, SLOT(reject()));
+
+    QTreeWidgetItem * pTreeWidgetItem_header = pTreeWidget_specialExits->headerItem();
+
+    pTreeWidgetItem_header->setToolTip(0, QStringLiteral("<html><head/><body>%1</body></html>")
+                                       .arg(tr("<p>Indicates if there is already a custom line for this special exit, which will be <b>replaced</b> if the exit is selected.</p>")));
+    pTreeWidgetItem_header->setToolTip(1, QStringLiteral("<html><head/><body>%1</body></html>")
+                                       .arg(tr("<p>The room this special exit leads <b>to</b>.</p>")));
+    pTreeWidgetItem_header->setToolTip(2, QStringLiteral("<html><head/><body>%1</body></html>")
+                                       .arg(tr("<p>The command or Lua script that goes to the given room.</p>"
+                                               "<p>It is what Mudlet will use to speedwalk (route find) using this particular exit so it can be a complex script - however <i>currently</i> Mudlet also uses this entry to identify this particular exit so if it is to be a <i>scripted</i> procedure to use this exit, a Lua function to do so is highly advisable!</p>")));
```

Comment:
I don't think many people will understand what this means.

Reply:
Currently we do not provide a means to have a name for a special exit that is an alias for a script - I recall that there is a comment somewhere in the C++ code for special exits that _"...a script should be entered with a `script: ` (or was it a `lua:`) prefix..."_ but that is confusing and wrong in that __some__ lua packages use that convention but it is not something that is hard coded in the C++ core.  It also means that if the commands to use a special exit are something like `unlock trapdoor; climb down; lock trapdoor` then THAT is the identifier that has to be used/displayed in the GUI associated with the mapper - however I would like to see that changed so that there can be an identifier `trapdoor` that can be used for display purposes associated with the original `command`...

... so at some future stage I hope to be able to change this text! 😀

Reply:
How about the following text:

> Mudlet will use this command in walking through this exit: it can be something simple as a direction or something more capable like a Lua script.

Reply:
☑️  Sentence emended to: "Mudlet will use this command in walking through this exit (when requested to find a route through the map); it can be something simple as a direction or something more capable like a Lua script."